Source/Urho3D/Network/Network.cpp

```diff
@@ -200,40 +326,69 @@ void Network::ClientDisconnected(kNet::MessageConnection* connection)
     }
 }
 
+void Network::SetDiscoveryBeacon(const VariantMap& data)
+{
+    VectorBuffer buffer;
+    buffer.WriteVariantMap(data);
+    if (buffer.GetSize() > 400)
+        URHO3D_LOGERROR("Discovery beacon of size: " + String(buffer.GetSize()) + " bytes is too large, modify MAX_OFFLINE_DATA_LENGTH in RakNet or reduce size");
+    rakPeer_->SetOfflinePingResponse((const char*)buffer.GetData(), buffer.GetSize());
+}
+
+void Network::DiscoverHosts(unsigned port)
+{
+    // JSandusky: Contrary to the manual, we actually do have to perform Startup first before we can Ping
+    if (rakPeer_ && !rakPeer_->IsActive())
+    {
+        SLNet::SocketDescriptor socket;
+        rakPeer_->Startup(1, &socket, 1);

```

Comment:
It'd be nice to have comments for magic numbers here and below, like `/* very important stuff */`